cmd/cluster/aws/create.go

```diff
@@ -27,99 +28,52 @@ import (
-	cmd.Flags().StringVar(&opts.ControlPlaneAvailabilityPolicy, "control-plane-availability-policy", opts.ControlPlaneAvailabilityPolicy, "Availability policy for hosted cluster components. Supported options: SingleReplica, HighlyAvailable")
-	cmd.Flags().StringVar(&opts.InfrastructureAvailabilityPolicy, "infra-availability-policy", opts.InfrastructureAvailabilityPolicy, "Availability policy for infrastructure services in guest cluster. Supported options: SingleReplica, HighlyAvailable")
-	cmd.Flags().StringSliceVar(&opts.AdditionalTags, "additional-tags", opts.AdditionalTags, "Additional tags to set on AWS resources")
-	cmd.Flags().BoolVar(&opts.GenerateSSH, "generate-ssh", opts.GenerateSSH, "If true, generate SSH keys")
-	cmd.Flags().StringVar(&opts.EtcdStorageClass, "etcd-storage-class", opts.EtcdStorageClass, "The persistent volume storage class for etcd data volumes")
+	cmd.Flags().StringVar(&opts.AWSPlatform.ControlPlaneOperatorImage, "control-plane-operator-image", opts.AWSPlatform.ControlPlaneOperatorImage, "Override the default image used to deploy the control plane operator")
+	cmd.Flags().StringVar(&opts.AWSPlatform.AWSCredentialsFile, "aws-creds", opts.AWSPlatform.AWSCredentialsFile, "Path to an AWS credentials file (required)")
+	cmd.Flags().StringVar(&opts.AWSPlatform.SSHKeyFile, "ssh-key", opts.AWSPlatform.SSHKeyFile, "Path to an SSH key file")
+	cmd.Flags().Int32Var(&opts.AWSPlatform.NodePoolReplicas, "node-pool-replicas", opts.AWSPlatform.NodePoolReplicas, "If >-1, create a default NodePool with this many replicas")
+	cmd.Flags().StringVar(&opts.AWSPlatform.InfrastructureJSON, "infra-json", opts.AWSPlatform.InfrastructureJSON, "Path to file containing infrastructure information for the cluster. If not specified, infrastructure will be created")
+	cmd.Flags().StringVar(&opts.AWSPlatform.IAMJSON, "iam-json", opts.AWSPlatform.IAMJSON, "Path to file containing IAM information for the cluster. If not specified, IAM will be created")
+	cmd.Flags().StringVar(&opts.AWSPlatform.Region, "region", opts.AWSPlatform.Region, "Region to use for AWS infrastructure.")
+	cmd.Flags().StringVar(&opts.AWSPlatform.InfraID, "infra-id", opts.AWSPlatform.InfraID, "Infrastructure ID to use for AWS resources.")
+	cmd.Flags().StringVar(&opts.AWSPlatform.InstanceType, "instance-type", opts.AWSPlatform.InstanceType, "Instance type for AWS instances.")
+	cmd.Flags().StringVar(&opts.AWSPlatform.BaseDomain, "base-domain", opts.AWSPlatform.BaseDomain, "The ingress base domain for the cluster")
+	cmd.Flags().StringArrayVar(&opts.AWSPlatform.Annotations, "annotations", opts.AWSPlatform.Annotations, "Annotations to apply to the hostedcluster (key=value). Can be specified multiple times.")
+	cmd.Flags().BoolVar(&opts.AWSPlatform.FIPS, "fips", opts.AWSPlatform.FIPS, "Enables FIPS mode for nodes in the cluster")
```

Comment:
Why are ControlPlaneOperatorImage, Annotations and EtcdStorageClass AWS specific?

Also I think this model should honour the API exposure, therefore:
FIPS 
SSHKeyFile
AutoRepair
NodePoolReplicas

should be core. There might be exceptions where they don't apply and so we should do proper API validation and fail/signal incompatibility appropriately, but I think it'll be practical to avoid duplication and we'll satisfy majority of the cases.

Reply:
> NodePoolReplicas

This one, I was unsure: `NodePoolReplicas` should probably be hardcoded to `-1` for `None`?

Reply:
Let's just keep the behaviour consistent for now and let the backend react appropriately. Yeh eventually there might be an agnostic implementation for None.